client/tests/integration/smartcontracts/executor_with_custom_permission/src/lib.rs

```diff
@@ -16,44 +16,18 @@ extern crate alloc;
 #[cfg(not(test))]
 extern crate panic_halt;
 
-use alloc::string::String;
-
-use anyhow::anyhow;
-use iroha_executor::{prelude::*, DataModelBuilder};
-use iroha_schema::IntoSchema;
+use executor_custom_data_model::permissions::CanControlDomainLives;
+use iroha_executor::{
+    data_model::prelude::*, permission::ExecutorPermision as _, prelude::*, DataModelBuilder,
+};
+use iroha_executor_data_model::permission::domain::CanUnregisterDomain;
 use lol_alloc::{FreeListAllocator, LockedAllocator};
-use parity_scale_codec::{Decode, Encode};
-use serde::{Deserialize, Serialize};
-use serde_json::json;
 
 #[global_allocator]
 static ALLOC: LockedAllocator<FreeListAllocator> = LockedAllocator::new(FreeListAllocator::new());
 
 getrandom::register_custom_getrandom!(iroha_executor::stub_getrandom);
 
-use alloc::format;
-
-mod token {
-    //! Module with custom token.
-
-    use super::*;
-
-    /// Token to identify if user can (un-)register domains.
-    #[derive(
-        PartialEq,
-        Eq,
-        Permission,
-        ValidateGrantRevoke,
```

Comment:
interestingly, this used to be derived but wasn't working at all because custom permission is not part of `AnyPermission` enum

Reply:
custom permission was deriving `OnlyGenesis` but was being registered on executor upgrade :)